src/libcore/mem.rs

```diff
@@ -143,6 +143,19 @@ pub fn forget<T>(t: T) {
     ManuallyDrop::new(t);
 }
 
+/// Like [`forget`], but also accepts unsized values.
+///
+/// This function is just a shim intended to be removed when the `unsized_locals` feature gets
+/// stabilized.
+///
+/// [`forget`]: fn.forget.html
+#[inline]
+#[cfg(not(stage0))]
+#[unstable(feature = "forget_unsized", issue = "0")]
+pub fn forget_unsized<T: ?Sized>(t: T) {
+    unsafe { intrinsics::forget(t) }
```

Comment:
Could `ManuallyDrop` be used in the future?

Reply:
Yes, provided we implement support for unsized return values. The reason is that `ManuallyDrop::new(t)` returns a `ManuallyDrop<T>`, which is unsized.

Reply:
Ah, I see, I was thinking in terms of construction. A `pub(crate)` `ManuallyDrop::forget` method that doesn't actually return `ManuallyDrop` could still work, right?

Reply:
Oh wait, you're totally right. I think adding the `forget` intrinsic was unnecessary since we can do:

```rust
pub fn forget_unsized<T: ?Sized>(t: T) {
    let _ = ManuallyDrop { value: t };
}
```

Reply:
So should we `r-` then and use that instead? Seems nicer than an intrinsic.

Reply:
It seems I was wrong. It's still impossible to initialize structs with unsized fields, so we'll have to live with `intrinsics::forget` for a while.

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2015&gist=6becf5139279bf16437a35617dfbf47b